*: add notice on flashback cluster usage (#16738)
pingcap/docs: sql-statements/sql-statement-flashback-cluster.md

>
> The `FLASHBACK CLUSTER TO [TIMESTAMP|TSO]` syntax is not applicable to [TiDB Serverless](https://docs.pingcap.com/tidbcloud/select-cluster-tier#tidb-serverless) clusters. To avoid unexpected results, do not execute this statement on TiDB Serverless clusters.

> **Warning:**
>

> > When using TSO to specify a recovery point in time, make sure to check the validity of the TSO and avoid specifying a time in the future that exceeds the [current TSO of TiDB](/tso.md). Otherwise, concurrent processing linear consistency and transaction isolation levels might be violated, leading to serious data correctness issues.

Author: ```suggestion
> When specifying a recovery point in time, make sure to check the validity of your specified timestamp or TSO and avoid specifying a future time that exceeds the maximum TSO currently allocated by the PD (see `Current TSO` on the Grafana PD panel). Otherwise, concurrent processing linear consistency and transaction isolation levels might be violated, leading to serious data correctness issues.
```

> > When specifying a recovery point in time, make sure to check the validity of your target timestamp or TSO and avoid specifying a future time that exceeds the maximum TSO currently allocated by the PD (see `Current TSO` on the Grafana PD panel). Otherwise, concurrent processing linear consistency and transaction isolation levels might be violated, leading to serious data correctness issues.

Author: ```suggestion
> When specifying a recovery point in time, make sure to check the validity of your target timestamp or TSO and avoid specifying a future time that exceeds the maximum TSO currently allocated by PD (see `Current TSO` on the Grafana PD panel). Otherwise, concurrent processing linear consistency and transaction isolation levels might be violated, leading to serious data correctness issues.
```

> > When specifying a recovery point in time, make sure to check the validity of your specified timestamp or TSO and avoid specifying a future time that exceeds the maximum TSO currently allocated by the PD (see `Current TSO` on the Grafana PD panel). Otherwise, concurrent processing linear consistency and transaction isolation levels might be violated, leading to serious data correctness issues.

Author: ```suggestion
> When specifying a recovery point in time, make sure to check the validity of your target timestamp or TSO and avoid specifying a future time that exceeds the maximum TSO currently allocated by the PD (see `Current TSO` on the Grafana PD panel). Otherwise, concurrent processing linear consistency and transaction isolation levels might be violated, leading to serious data correctness issues.
```
